### PR TITLE
Distributed

### LINKED ISSUE #100: Distributed TensorFlow
* Markdown文件：[文件地址](https://github.com/xitu/tensorflow/blob/zh-hans/tensorflow/docs_src/deploy/distributed.md)
  * PR 地址：#148

  ----
  * 翻译时间：5 天
  * 校对时间：1 天
  * 翻译积分：5 分
  * 校对积分：1.5 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by filosfino**
resolve #100

### CODE REVIEW

**Comment by jaymz1439** [comment]
认领校对

**Comment by wzy816** [comment]
校对认领

File: tensorflow/docs_src/deploy/distributed.md [hunk 1]
```diff
@@ -141,52 +109,25 @@ with tf.Session("grpc://worker7.example.com:2222") as sess:
...
-  [CIFAR-10 multi-GPU trainer](https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10/cifar10_multi_gpu_train.py)),
-  and between-graph replication (e.g. using the
-  @{tf.train.SyncReplicasOptimizer}).
-
-### Putting it all together: example trainer program
-
-The following code shows the skeleton of a distributed trainer program,
-implementing **between-graph replication** and **asynchronous training**. It
-includes the code for the parameter server and worker tasks.
+在上面的例子中，变量是在 `ps` 作业中的两个任务上创建的，模型的计算密集部分是在 `worker` 作业中创建的。TensorFlow 将在作业之间插入适当的数据传输（正向传递时从 `ps` 到 `worker`，应用导数从 `worker` 到 `ps`）。
```
Comment (wzy816) [inline tensorflow/docs_src/deploy/distributed.md]: gradient 是否翻译成“梯度更好”？

File: tensorflow/docs_src/deploy/distributed.md [hunk 2]
```diff
@@ -141,52 +109,25 @@ with tf.Session("grpc://worker7.example.com:2222") as sess:
...
+
+* **图间复制** 在这种方法中，有一个单独的客户端对于每个 `/job:worker` 任务，通常与 worker 任务位于相同进程中。每个客户端构建出一个类似的包含参数（像以往一样，通过 @{tf.train.replica_device_setter} 确定性地映射，固定对应到相同的任务 `/job:ps`）的图; 和一个模型中的计算密集型部分的单一的副本，固定对应到本地任务中 `/job:worker`。
+
+* **异步训练** 在这种方法中，图的每个副本都有一个没有独立训练循环，不做协调就可以执行。它是兼容的以上两种形式的复制。
+
+* **同步训练** 在这种方法中，所有的副本读取到一致的的当前参数值，并行计算梯度，然后将它们一起应用。它与图内复制（例如：像[CIFAR-10 multi-GPU trainer](https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10/cifar10_multi_gpu_train.py)一样使用梯度平均和多 GPU 图间复制），图间复制（使用 @{tf.train.SyncReplicasOptimizer}）。
+
+### 总结：示例训练程序
+
+以下代码显示了分布式训练程序的框架，实现**图间复制**和**异步培训**。它包括参数服务器和 `Worker` 任务的代码。
```
Comment (wzy816) [inline tensorflow/docs_src/deploy/distributed.md]: 异步培训 => 异步训练
与上文一致

File: tensorflow/docs_src/deploy/distributed.md [hunk 3]
```diff
@@ -213,40 +154,38 @@ def main(_):
...
 
-    # The StopAtStepHook handles stopping after running given steps.
+    # StopAtStepHook 在运行给定步骤后处理停止
     hooks=[tf.train.StopAtStepHook(last_step=1000000)]
 
-    # The MonitoredTrainingSession takes care of session initialization,
-    # restoring from a checkpoint, saving to a checkpoint, and closing when done
-    # or an error occurs.
+    # MonitoredTrainingSession 负责会话初始化
+    # 从检查点恢复，保存到检查点，完成后关闭或报错
```
Comment (wzy816) [inline tensorflow/docs_src/deploy/distributed.md]: 完成后关闭或报错 =>  一旦完成或报错就关闭

File: tensorflow/docs_src/deploy/distributed.md [hunk 4]
```diff
@@ -213,40 +154,38 @@ def main(_):
...
                                            checkpoint_dir="/tmp/train_logs",
                                            hooks=hooks) as mon_sess:
       while not mon_sess.should_stop():
-        # Run a training step asynchronously.
-        # See `tf.train.SyncReplicasOptimizer` for additional details on how to
-        # perform *synchronous* training.
-        # mon_sess.run handles AbortedError in case of preempted PS.
+        # 异步运行训练
+        # 有关如何执行同步训练的更多信息，请参见 `tf.train.SyncReplicasOptimizer`
+        # mon_sess.run在被抢占PS的情况下处理AbortedError。
```
Comment (wzy816) [inline tensorflow/docs_src/deploy/distributed.md]: 中英文之间缺空格

File: tensorflow/docs_src/deploy/distributed.md [hunk 5]
```diff
@@ -304,51 +242,33 @@ $ python trainer.py \
...
-to a single process. A task belongs to a particular "job" and is identified by
-its index within that job's list of tasks.
 
-**TensorFlow server** A process running
-a @{tf.train.Server} instance, which is
-a member of a cluster, and exports a "master service" and "worker service".
+**Worker 服务**
 
-**Worker service**
+一个使用本地设备执行 TensorFlow 计算图中部分命令的 RPC 服务。工人服务实现了[worker_service.proto](https://www.tensorflow.org/code/tensorflow/core/protobuf/worker_service.proto)。所有的 TensorFlow 服务器都实现了 Worker 服务。
```
Comment (wzy816) [inline tensorflow/docs_src/deploy/distributed.md]: 工人服务 => worker 服务
worker可以统一保持不翻译

File: tensorflow/docs_src/deploy/distributed.md [hunk 6]
```diff
@@ -82,43 +65,28 @@ tf.train.ClusterSpec({
...
 server = tf.train.Server(cluster, job_name="local", task_index=1)
 ```
 
-**Note:** Manually specifying these cluster specifications can be tedious,
-especially for large clusters. We are working on tools for launching tasks
-programmatically, e.g. using a cluster manager like
-[Kubernetes](http://kubernetes.io). If there are particular cluster managers for
-which you'd like to see support, please raise a
-[GitHub issue](https://github.com/tensorflow/tensorflow/issues).
+**注意：** 手动指定这些集群规范可能很乏味，特别是对于大型集群。我们正在开发可编程的任务启动工具，例如类似 [Kubernetes](http://kubernetes.io) 的集群管理器。如果你有希望获得支持的集群管理器请提出一个 [GitHub issue](https://github.com/tensorflow/tensorflow/issues)。
```
[the remark this answers is not in the document]
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 感谢：）的确这样翻更通顺

File: tensorflow/docs_src/deploy/distributed.md (hunk 1 above)
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 是的，还有`应用`这个词我觉得也是表达不准确，是否直接使用`反向传递梯度下降`比较好？

**Comment by leviding** [comment]
@jaymz1439 记得来校对哈

File: tensorflow/docs_src/deploy/distributed.md [hunk 7]
```diff
@@ -1,62 +1,45 @@
...
 
-## Hello distributed TensorFlow!
+## 你好，分布式TensorFlow！
 
-To see a simple TensorFlow cluster in action, execute the following:
+要查看一个简单的 TensorFlow 集群，请执行以下操作：
 
 ```shell
-# Start a TensorFlow server as a single-process "cluster".
+# 启动一个TensorFlow服务器作为一个单一进程的“集群”
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 启动一个单一进程“集群”的TensorFlow服务器。

File: tensorflow/docs_src/deploy/distributed.md [hunk 8]
```diff
@@ -1,62 +1,45 @@
...
->>> sess = tf.Session(server.target)  # Create a session on the server.
+>>> sess = tf.Session(server.target)  # 在服务器上创建一个会话。
 >>> sess.run(c)
 'Hello, distributed TensorFlow!'
 ```
 
-The
 @{tf.train.Server.create_local_server}
-method creates a single-process cluster, with an in-process server.
+方法使用进程内服务器创建了一个单进程群集。
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 在运行的服务器上创建一个单进程的集群的方法

File: tensorflow/docs_src/deploy/distributed.md [hunk 9]
```diff
@@ -141,52 +109,25 @@ with tf.Session("grpc://worker7.example.com:2222") as sess:
...
+
+## 重复训练
+
+一种通用训练的配置，也被称为“并行数据”，包含了使用不同 mini-batch 来训练相同模型的`Worker`作业中的多个任务，更新`ps`作业中一个或多个任务里的共享参数。所有任务通常在不同的机器上运行。在 TensorFlow 中有很多方法可以指定任务分配的结构，我们正在开发简化指定复制模型工作的库。可能的方法包括：
+
+* **图内复制** 在这种方法中，客户端构建一个包含一组参数（在 `tf.Variable` 节点上固定
+   到 `/job:ps`）的 `tf.Graph`; 以及模型的计算密集型部分的多个副本，
+   每个副本固定对应到 `/job:worker` 中不同的任务上。
+
+* **图间复制** 在这种方法中，有一个单独的客户端对于每个 `/job:worker` 任务，通常与 worker 任务位于相同进程中。每个客户端构建出一个类似的包含参数（像以往一样，通过 @{tf.train.replica_device_setter} 确定性地映射，固定对应到相同的任务 `/job:ps`）的图; 和一个模型中的计算密集型部分的单一的副本，固定对应到本地任务中 `/job:worker`。
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 在这种方法中，对于每个`/job:worker`任务会有一个独立的客户端，通常把相同进程放在一个worker任务中。

File: tensorflow/docs_src/deploy/distributed.md (hunk 8 above)
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 这样译和前面@连起来看的话不太顺吧

File: tensorflow/docs_src/deploy/distributed.md [hunk 10]
```diff
@@ -141,52 +109,25 @@ with tf.Session("grpc://worker7.example.com:2222") as sess:
...
+
+* **异步训练** 在这种方法中，图的每个副本都有一个没有独立训练循环，不做协调就可以执行。它是兼容的以上两种形式的复制。
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 方法改为方式好点，个人意见

File: tensorflow/docs_src/deploy/distributed.md [hunk 11]
```diff
@@ -141,52 +109,25 @@ with tf.Session("grpc://worker7.example.com:2222") as sess:
...
+
+* **同步训练** 在这种方法中，所有的副本读取到一致的的当前参数值，并行计算梯度，然后将它们一起应用。它与图内复制（例如：像[CIFAR-10 multi-GPU trainer](https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10/cifar10_multi_gpu_train.py)一样使用梯度平均和多 GPU 图间复制），图间复制（使用 @{tf.train.SyncReplicasOptimizer}）。
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 所有的副本读取相同的值赋给当前的参数

File: tensorflow/docs_src/deploy/distributed.md [hunk 12]
```diff
@@ -304,51 +242,33 @@ $ python trainer.py \
 
 ## Glossary
 
-**Client**
+**客户端**
+
+客户端通常是一个构建 TensorFlow 计算图并构建一个 `tensorflow::Session` 与群集进行交互。通常用 Python 或 C++ 编写。一个客户端进程可以直接与多个 TensorFlow 服务器交互（参阅上面的“重复训练”），一台服务器可为多个客户端服务。
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 客户端是一段程序，用来构建TensorFlow图表和用于与集群交互的会话`tensorflow::Session`。

File: tensorflow/docs_src/deploy/distributed.md [hunk 13]
```diff
@@ -304,51 +242,33 @@ $ python trainer.py \
...
 
-A client is typically a program that builds a TensorFlow graph and constructs a
-`tensorflow::Session` to interact with a cluster. Clients are typically written
-in Python or C++. A single client process can directly interact with multiple
-TensorFlow servers (see "Replicated training" above), and a single server can
-serve multiple clients.
+**作业**
 
-**Cluster**
+一份作业包括一份“任务”清单，通常用于一个共同的目的。例如，名为 `ps`的作业（对于“参数服务器”）通常包括存储和更新变量的节点; 而名为 `worker` 的作业通常包括执行计算密集型任务的无状态节点。工作中的任务通常运行在不同的机器上。这套工作角色是灵活的：例如，`Worker` 可能会保持某种状态。
```
Comment (jaymz1439) [inline tensorflow/docs_src/deploy/distributed.md]: 名为`ps`的作业(“参数服务器”)管理存储和更新变量的节点。这样翻是不是好点

**Comment by leviding** [comment]
@jaymz1439 校对完成了吗

**Comment by jaymz1439** [comment]
完成了 @leviding

**Comment by leviding** [comment]
@filosfino 可以修改啦

File: tensorflow/docs_src/deploy/distributed.md (hunk 13 above)
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 作业应该不负责管理节点吧？在我理解中作业应该是在某些节点上发生，可以类比为这些节点是作业的宿主。用`包括`可能缺少了宿的概念，但一时找不到合适的词

File: tensorflow/docs_src/deploy/distributed.md (hunk 12 above)
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 同意

File: tensorflow/docs_src/deploy/distributed.md (hunk 11 above)
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 同意

File: tensorflow/docs_src/deploy/distributed.md (hunk 9 above)
Comment (filosfino) [reply tensorflow/docs_src/deploy/distributed.md]: 同意

**Comment by filosfino** [comment]
@leviding 修改完毕

**Comment by leviding** [comment]
@filosfino @jaymz1439 @wzy816 大家辛苦啦，新年快乐 😊有空的话还有两个校对可以认领下哈。
